workspaces/src/types/mod.rs

```diff
@@ -128,6 +129,19 @@ impl PublicKey {
         Self(near_crypto::PublicKey::empty(key_type.into_near_keytype()))
     }
 
+    /// Create a new [`PublicKey`] from the given bytes. This will return an error if the bytes are not in the
+    /// correct format. Expected to have key type be the first byte encoded, with the remaining bytes being the
+    /// key data.
+    pub fn try_from_bytes(bytes: &[u8]) -> Result<Self> {
```

Comment:
I think we should reduce the scope of exposure here, if there's no way to generate a slice from `workspaces::PublicKey` that this method can then parse, we probably shouldn't have this method public.

```suggestion
    pub(crate) fn try_from_bytes(bytes: &[u8]) -> Result<Self> {
```

Reply:
I'm on the fence with keeping or not keeping it, but I was thinking we'd have a raw bytes API still so that people could construct it if they have the info from a different source than just workspaces or sdk. `BorshDeserialize::try_from_slice` would probably not be intuitive since we can't attach any docs noting that it requires 4 extra bytes denoting the length at the start.

Reply:
> but I was thinking we'd have a raw bytes API still so that people could construct it if they have the info from a different source than just workspaces or sdk.

So the input here is the borsh-serialized format of `near-crypto::PublicKey`.

So they either already have a `near_crypto::PublicKey` or they manually constructed the slice.

- For the first, perhaps we just add a `From<near_crypto::PublicKey>` for `workspaces::PublicKey`?
- For the second, this signature could be `pub fn from_parts(key_type: KeyType, key_data: &[u8]) -> Result<Self>`

What do you think?

Reply:
Why do we need a copy-pasted implementation of `KeyType` in the first place? Wouldn't `near_crypto` types work here? (we already have it as a dependency)

Reply:
it's because we don't want to expose it as a transient dependency quiet yet, since that would lead us to maintaining that relationship going forward and cause unwanted breaking changes along the way if `near_crypto` were to change. So, until `near_crypto` is stable (i.e. 1.0), then we shouldn't expose it and just maintain a version ourselves to have this level of control. This was also done for things like `near_primitives` as well to avoid that hassle

Reply:
@ChaoticTempest Thanks for the context!